### ISSUE TITLE
Appsignal fails to compile because download of libappsignal.a fails.

### ISSUE THREAD

**Issue by idyll** [appsignal/appsignal-elixir#4]
```
 cc: error: /app/_build/prod/lib/appsignal/priv/libappsignal.a: No such file or directory
```

It looks like the file fails to download which in turn causes the build to fail. (Which in turn causes my docker image not to compile... and so on.)

I would say it happens about once a day or so and we deploy about 16 times a day. So maybe 1/16.
It's not reproducible.

Running the same build again will succeed. It never fails twice as far as I have seen.

**Comment by thijsc** [comment]
My first guess would be that the CDN sometimes fails. It looks like we don't have error logging for this currently: https://github.com/appsignal/appsignal-elixir/blob/master/mix_helpers.exs#L21

**Comment by arjan** [comment]
I'll add error logging and a retry loop to try it a couple of times and then fail with an appropriate message.

**Comment by arjan** [comment]
I just published version [0.0.8](https://hex.pm/packages/appsignal); the file download is now retried 5 times (with backoff, using curl's `--retry` argument). On top of that , checksum verification of the archive is now implemented. (see 524ca210dbdb314123e75819578a5a78f02d572b)

Also, when the agent file already has been downloaded, it is not downloaded again (see ed0c01835b0fc2aeb7700128f7445d0f08017f8b). This should also speed up your builds.
